### ISSUE TITLE
getting extraneous errors when `npm ls -g`

### ISSUE THREAD

**Issue by dfuentes77** [tj/n#440]
### Description
Latest OSX Sierra 10.12.5
n -V output: 2.1.7
Installed via n-install

### I have these versions installed

``` 
    node/6.10.3
  ο node/7.9.0
    node/8.0.0
```

I run `npm ls --depth=0 -g`

and I get

```
/Users/<username>/.n/lib
└── npm@4.2.0

npm ERR! extraneous: call-limit@1.1.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/call-limit
npm ERR! extraneous: detect-indent@5.0.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/detect-indent
npm ERR! extraneous: lazy-property@1.0.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/lazy-property
npm ERR! extraneous: pacote@2.7.21 /Users/<username>/.n/lib/node_modules/npm/node_modules/pacote
npm ERR! extraneous: update-notifier@2.1.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/update-notifier
npm ERR! extraneous: path-array@1.0.1 /Users/<username>/.n/lib/node_modules/npm/node_modules/node-gyp/node_modules/path-array
npm ERR! extraneous: has-color@0.1.7 /Users/<username>/.n/lib/node_modules/npm/node_modules/npmlog/node_modules/gauge/node_modules/has-color
npm ERR! extraneous: number-is-nan@1.0.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/npmlog/node_modules/gauge/node_modules/string-width/node_modules/code-point-at/node_modules/number-is-nan
npm ERR! extraneous: bl@1.1.2 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/bl
npm ERR! extraneous: node-uuid@1.4.7 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/node-uuid
npm ERR! extraneous: performance-now@0.2.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/performance-now
npm ERR! extraneous: safe-buffer@5.0.1 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/safe-buffer
npm ERR! extraneous: ajv@4.11.8 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/har-validator/node_modules/ajv
npm ERR! extraneous: har-schema@1.0.5 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/har-validator/node_modules/har-schema
npm ERR! extraneous: assert-plus@1.0.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/request/node_modules/http-signature/node_modules/jsprim/node_modules/assert-plus
npm ERR! extraneous: stream-shift@1.0.0 /Users/<username>/.n/lib/node_modules/npm/node_modules/sorted-union-stream/node_modules/stream-iterate/node_modules/stream-shift
```

**Comment by dfuentes77** [comment]
After installing AVN I now get:

```
$ npm ls --depth=0 -g
/Users/<username>/.n/lib
├── avn@0.2.3
├── avn-n@0.2.0
├── avn-nvm@0.2.1
└── npm@5.0.0

npm ERR! missing: npm-package-arg@^4.1.1, required by realize-package-specifier@3.0.3
npm ERR! invalid: glob@6.0.4 /Users/<username>/.n/lib/node_modules/npm/node_modules/init-package-json/node_modules/glob
npm ERR! invalid: once@1.3.3 /Users/<username>/.n/lib/node_modules/npm/node_modules/mississippi/node_modules/end-of-stream/node_modules/once
npm ERR! invalid: readable-stream@2.0.6 /Users/<username>/.n/lib/node_modules/npm/node_modules/npm-registry-client/node_modules/concat-stream/node_modules/readable-stream
npm ERR! invalid: glob@6.0.4 /Users/<username>/.n/lib/node_modules/npm/node_modules/read-package-json/node_modules/glob
```

**Comment by troy0820** [comment]
Your npm install is at /Users/<username>/.n/lib  

I would recommend using `npm install -g n` to install n.  The n-install doesn't default the install dir to /usr/local/lib like the npm install.  I would check that and see if that fixes your problem.

**Comment by dfuentes77** [comment]
So on Mac, what base install of node/npm would you recommend in order to install `n` since I think by default it doesn't exist.  It's kind of a weird catch22 right?  I need node/npm to install `n`, a node version manager, to install different versions of node/npm ;)

**Comment by troy0820** [comment]
Install it with `https://nodejs.org/en/`.  That'll give you npm as well, and then you can install the node version manager `n` with `npm i -g n`

**Comment by goldenyz** [comment]
have the same issue and installed with npm i -g n:
```
/usr/local/lib
├── fie@2.7.7
├── http-server@0.10.0
├── n@2.1.8
├── npm@3.10.10
├── qap-cli@3.1.1
├── tnpm@4.24.1
└── weex-toolkit@1.0.5

npm ERR! extraneous: abbrev@1.1.0 /usr/local/lib/node_modules/fie/node_modules/_abbrev@1.1.0@abbrev
npm ERR! extraneous: address@1.0.2 /usr/local/lib/node_modules/fie/node_modules/_address@1.0.2@address
npm ERR! extraneous: agent-base@2.1.1 /usr/local/lib/node_modules/fie/node_modules/_agent-base@2.1.1@agent-base
```

**Comment by dfuentes77** [comment]
@troy0820 @goldenyz Same here,  I just uninstalled my whole n/avn setup and installed [LTS node from official site](https://nodejs.org/en/) (which btw, always requires sudo since it's installed as root) and I installed `n` and `avn` as recommended, and I still get similar output:

```
$ sudo npm ls --depth=0 -g
Password:
/usr/local/lib
├── avn@0.2.3
├── avn-n@0.2.0
├── avn-nvm@0.2.1
├── n@2.1.8
└── npm@4.2.0

npm ERR! extraneous: call-limit@1.1.0 /usr/local/lib/node_modules/npm/node_modules/call-limit
npm ERR! extraneous: detect-indent@5.0.0 /usr/local/lib/node_modules/npm/node_modules/detect-indent
npm ERR! extraneous: lazy-property@1.0.0 /usr/local/lib/node_modules/npm/node_modules/lazy-property
npm ERR! extraneous: libnpx@9.2.1 /usr/local/lib/node_modules/npm/node_modules/libnpx
npm ERR! extraneous: pacote@2.7.38 /usr/local/lib/node_modules/npm/node_modules/pacote
npm ERR! extraneous: worker-farm@1.4.1 /usr/local/lib/node_modules/npm/node_modules/worker-farm
npm ERR! extraneous: path-array@1.0.1 /usr/local/lib/node_modules/npm/node_modules/node-gyp/node_modules/path-array
npm ERR! extraneous: has-color@0.1.7 /usr/local/lib/node_modules/npm/node_modules/npmlog/node_modules/gauge/node_modules/has-color
npm ERR! extraneous: number-is-nan@1.0.0 /usr/local/lib/node_modules/npm/node_modules/npmlog/node_modules/gauge/node_modules/string-width/node_modules/code-point-at/node_modules/number-is-nan
npm ERR! extraneous: bl@1.1.2 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/bl
npm ERR! extraneous: node-uuid@1.4.7 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/node-uuid
npm ERR! extraneous: performance-now@0.2.0 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/performance-now
npm ERR! extraneous: ajv@4.11.8 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/har-validator/node_modules/ajv
npm ERR! extraneous: har-schema@1.0.5 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/har-validator/node_modules/har-schema
npm ERR! extraneous: assert-plus@1.0.0 /usr/local/lib/node_modules/npm/node_modules/request/node_modules/http-signature/node_modules/jsprim/node_modules/assert-plus
npm ERR! extraneous: stream-shift@1.0.0 /usr/local/lib/node_modules/npm/node_modules/sorted-union-stream/node_modules/stream-iterate/node_modules/stream-shift
npm ERR! invalid: ansi-regex@3.0.0 /usr/local/lib/node_modules/npm/node_modules/strip-ansi/node_modules/ansi-regex
```

**Comment by troy0820** [comment]
Extraneous packages _even though the error is there_ doesn't impact N from being functional.  This is because some of these packages are either invalid or not in your parent's package.json.  

You can read about it here [npm docs](https://docs.npmjs.com/cli/prune)

**Comment by JohnRGee** [comment]
I was having the same issue and narrowed down some reproduce steps. I think the `extraneous` messages related to npm/node_modules are because n is simply overwriting with each install and not removing older files. When the npm dependencies change between versions, the left-over files from previous npm installations cause the error messages from some commands.

Reproduce steps with default locations on OS X.:
```
n 8.7.0
rm -rf /usr/local/lib/node_modules/npm/node_modules
n 6.11.4
npm list --global --depth=0  # clean
n 8.7.0
npm list --global --depth=0  # extraneous
```

**Comment by troy0820** [comment]
Issue closed from merging PR #470.
